javascript/javascript1/week4/homework/Voice_Assistant.js

```diff
@@ -0,0 +1,154 @@
+
+Add fishing to my todo - Should respond with "fishing added to your todo". Should add fishing to a list of todos
+
+Add singing in the shower to my todo - Should add singing in the shower to a list of todos
+
+Remove fishing from my todo - Should respond with "Removed fishing from your todo"
+
+What is on my todo? - should respond with the todos. Fx you have 2 todos - fishing and singing in the shower
+
+What day is it today? - Should respond with the date in a human readable format. E.g. if today is 30/8/2019 then it should respond with 30. of August 2019
+
+Should be able to do simple math. fx what is 3 + 3 should respond with 6. Or what is 4 * 12 should respond with 48
+
+Set a timer for 4 minutes - Should respond with "Timer set for 4 minutes". When 4 minutes is up: "Timer done". How do we set a timer in js? Google is your friend here!*/
+
+
+let userName = ""; 
+let toDoArray = []; 
+let timerMinutes; 
+const millisecondsInAMinute = 60000;
+
+function getReply(command) {
+    if (command.startsWith("Hello my name is ")) {
+        const savedName = command.replace("Hello my name is ", ""); 
+
+        if (savedName === "") {  
+            return "Please tell me your name"; 
+        }
+
+        if (userName === savedName) {
+            return `You already told me your name is ${userName}!`;
+        }
+
+        userName = savedName;
+        return `Nice to meet you, ${userName}`;
+
+    } 
+
+    else if (command === "What is my name?") { 
+        if (userName) {
+            return `Your name is ${userName}`;
+        } else {
+            return "I don't know your name.";
+        }
+    }
+    
+    if (command === "Add fishing to my todo") {
+        toDoArray.push("fishing");
+        return "Fishing added to your todo.";
+    }
+
+    if (command === "Add singing in the shower to my todo") {
+        toDoArray.push("singing in the shower");
+        return "Singing in the shower added to your todo.";
+    }
+
+    if (command === "Remove fishing from my todo") {
+        for (let i = 0; i < toDoArray.length; i++) {
+            if (toDoArray[i] === "fishing") { 
+                toDoArray.splice(i, 1);
+                return "Removed fishing from your todo."; 
+            }
+        }
+    }
+
+    if (command === "What is on my todo?") {
+        return toDoArray.join(", ");
+    }
+
+    if (command === "What day is it today?") {
+        let today = new Date();
+        let day = today.getDate();
+        let months = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
+        let month = months[today.getMonth()];
+        let year = today.getFullYear(); 
+
+        return `${day}. of ${month}. of ${year}`;
+    }
+
+
+    if (command.startsWith("What is ")) {
+        if (command.includes("+")) {
+            let numbers = command.replace("What is ", "").split(" + "); 
+            let sum = parseInt(numbers[0]) + parseInt(numbers[1]);
```

Comment:
Nice job here. 2 tips (you don't need to change your code)
- To avoid some repeated code, you could parse the numbers and the operator once at the top, then use a switch for every case.
- The other one is to use regexes, which are very powerful, but pretty hard to debug.